### PR TITLE
refactor: Rewrite empty_string & add more test cases

### LINKED ISSUE #6221: empty_string() needs a rewrite
**Describe the bug**

https://github.com/fossasia/open-event-server/blob/development/app/api/helpers/utilities.py#L28-L34 is exceedingly convoluted, does `__type(x)__` compares contrary to PEP8, and contains several logic errors.  A code review of this function would be a worthwhile effort.  Some `__doctests__` would really be helpful here.

**To Reproduce**

**Expected behavior**

**Stacktrace**

**Additional details (please complete the following information):**
 - OS: [e.g. MacOS, Ubuntu, CentOS]
 - Python Version [e.g. `3.5`, `3.6`]
 - `HEAD` Commit hash [e.g. `4629c62`]

**Additional context**

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Fixes #6221 

#### Short description of what this resolves:
Refactor helper function in accordance with python3 & add more test cases

#### Changes proposed in this pull request:

- Refactor empty_string
- Add more test cases for `empty_string`

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: tests/all/integration/api/helpers/test_utilities.py [hunk 1]
```diff
@@ -33,13 +33,10 @@ def test_string_empty(self):
         """Method to test whether an empty string is correctly identified."""
 
         with app.test_request_context():
-            self.assertTrue(string_empty(''))
-            self.assertTrue(string_empty(' '))
-            self.assertFalse(string_empty('some value'))
-            self.assertFalse(string_empty('  some   value '))
```
Comment (iamareebjamal) [inline tests/all/integration/api/helpers/test_utilities.py]: Let these test cases be there

Add test cases with tabs and newlines

Add test case with None

File: tests/all/integration/api/helpers/test_utilities.py [hunk 2]
```diff
@@ -33,13 +33,10 @@ def test_string_empty(self):
...
-            self.assertFalse(string_empty(str))
-            self.assertFalse(string_empty(int))
-            self.assertFalse(string_empty(None))
+            self.assertTrue(string_empty('  '))
+            self.assertFalse(string_empty(0))
+            self.assertFalse(string_empty([]))
+            self.assertFalse(string_empty(False))
```
Comment (iamareebjamal) [inline tests/all/integration/api/helpers/test_utilities.py]: Since this is a unit test, move to unit test file please

And open an issue to move all unit tests to the correct directory

Comment (mrsaicharan1) [reply tests/all/integration/api/helpers/test_utilities.py]: @iamareebjamal Yeah, I think that should be done right away.

**Comment by mrsaicharan1** [comment]
@iamareebjamal Moved to unit, restored test cases. Please have a look.

File: tests/all/unit/api/helpers/test_utilities.py [hunk 3]
```diff
@@ -17,6 +17,20 @@ def test_get_filename_from_cd(self):
...
         self.assertEqual(expected_response_none, response_none)
 
+    def test_string_empty(self):
+        """Method to test whether an empty string is correctly identified."""
+
+        self.assertTrue(string_empty(''))
+        self.assertTrue(string_empty(' '))
+        self.assertTrue(string_empty('  '))
+        self.assertTrue(string_empty('\n'))
+        self.assertFalse(string_empty(None))
```
Comment (iamareebjamal) [inline tests/all/unit/api/helpers/test_utilities.py]: Should be true IMO

File: tests/all/unit/api/helpers/test_utilities.py [hunk 4]
```diff
@@ -17,6 +17,20 @@ def test_get_filename_from_cd(self):
         self.assertEqual(expected_response_first, response_first)
         self.assertEqual(expected_response_none, response_none)
 
+    def test_string_empty(self):
+        """Method to test whether an empty string is correctly identified."""
+
+        self.assertTrue(string_empty(''))
+        self.assertTrue(string_empty(' '))
+        self.assertTrue(string_empty('  '))
+        self.assertTrue(string_empty('\n'))
```
Comment (iamareebjamal) [inline tests/all/unit/api/helpers/test_utilities.py]: Add tabs as well

Comment (mrsaicharan1) [reply tests/all/unit/api/helpers/test_utilities.py]: The third test case is a tab but will make it into multiple tabs

Comment (mrsaicharan1) [reply tests/all/unit/api/helpers/test_utilities.py]: Added

File: tests/all/unit/api/helpers/test_utilities.py (hunk 3 above)
Comment (mrsaicharan1) [reply tests/all/unit/api/helpers/test_utilities.py]: `None` isn't str object I believe?

Comment (iamareebjamal) [reply tests/all/unit/api/helpers/test_utilities.py]: A lot of times None can be passed for a string. We need to treat it as en empty string

Comment (mrsaicharan1) [reply tests/all/unit/api/helpers/test_utilities.py]: Travis was failing when I did that. What could be done?
